src/main/scala/com/github/codelionx/distod/actors/Worker.scala

```diff
@@ -0,0 +1,116 @@
+package com.github.codelionx.distod.actors
+
+import akka.actor.typed.{ActorRef, Behavior}
+import akka.actor.typed.scaladsl.{ActorContext, Behaviors}
+import com.github.codelionx.distod.Serialization.CborSerializable
+import com.github.codelionx.distod.actors.Master.DispatchWork
+import com.github.codelionx.distod.actors.Worker.{CheckCandidateNode, Command, WrappedPartitionEvent}
+import com.github.codelionx.distod.protocols.PartitionManagementProtocol._
+import com.github.codelionx.distod.types.{CandidateSet, PendingJobMap}
+
+
+object Worker {
+
+  sealed trait Command extends CborSerializable
+  final case class CheckCandidateNode(
+      candidateId: CandidateSet,
+      spiltCandidates: CandidateSet,
+      swapCandidates: Seq[(Int, Int)]
+  ) extends Command
+  private case class WrappedPartitionEvent(event: PartitionEvent) extends Command
+
+  def name(n: Int): String = s"worker-$n"
+
+  def apply(partitionManager: ActorRef[PartitionCommand], master: ActorRef[Master.Command]): Behavior[Command] =
+    Behaviors.setup[Command] { context =>
+      new Worker(context, master, partitionManager).start()
+    }
+
+}
+
+
+class Worker(
+    context: ActorContext[Command], master: ActorRef[Master.Command], partitionManager: ActorRef[PartitionCommand]
+) {
+
+  private val partitionEventMapper = context.messageAdapter(e => WrappedPartitionEvent(e))
+
+  def start(): Behavior[Command] = initialize()
+
+  def initialize(): Behavior[Command] = {
+    partitionManager ! LookupAttributes(partitionEventMapper)
+
+    Behaviors.receiveMessagePartial {
+      case WrappedPartitionEvent(AttributesFound(attributes)) =>
+        context.log.debug("Worker ready to process candidates: Master={}, Attributes={}", master, attributes)
+        master ! DispatchWork(context.self)
+        behavior(attributes)
+    }
+  }
+
+  def behavior(attributes: Seq[Int]): Behavior[Command] = Behaviors.receiveMessage {
+    case task @ CheckCandidateNode(candidateId, spiltCandidates, swapCandidates) =>
+      context.log.info("Checking candidate node {}", candidateId)
+
+      checkSplitCandidates(attributes, task)
+
+    case WrappedPartitionEvent(event) =>
+      context.log.info("Ignored {}", event)
+      Behaviors.same
+  }
+
+  def checkSplitCandidates(attributes: Seq[Int], task: CheckCandidateNode): Behavior[Command] = {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Method is longer than 50 lines](https://app.codacy.com/app/CodeLionX/distod/pullRequest?prid=4537008)